commit 79eb44d36e925c8ee65f4afaee4de8c6c5c12fa4
Author: Christophe Leroy <christophe.leroy@c-s.fr>
Date:   Wed Mar 1 15:33:26 2017 +0100

    netfilter: nf_conntrack_sip: fix wrong memory initialisation
    
    commit da2f27e9e615d1c799c9582b15262458da61fddc upstream.
    
    In commit 82de0be6862cd ("netfilter: Add helper array
    register/unregister functions"),
    struct nf_conntrack_helper sip[MAX_PORTS][4] was changed to
    sip[MAX_PORTS * 4], so the memory init should have been changed to
    memset(&sip[4 * i], 0, 4 * sizeof(sip[i]));
    
    But as the sip[] table is allocated in the BSS, it is already set to 0
    
    Fixes: 82de0be6862cd ("netfilter: Add helper array register/unregister functions")
    Signed-off-by: Christophe Leroy <christophe.leroy@c-s.fr>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/netfilter/nf_conntrack_sip.c b/net/netfilter/nf_conntrack_sip.c
index c3fc14e..3a8dc39 100644
--- a/net/netfilter/nf_conntrack_sip.c
+++ b/net/netfilter/nf_conntrack_sip.c
@@ -1630,8 +1630,6 @@ static int __init nf_conntrack_sip_init(void)
 		ports[ports_c++] = SIP_PORT;
 
 	for (i = 0; i < ports_c; i++) {
-		memset(&sip[i], 0, sizeof(sip[i]));
-
 		nf_ct_helper_init(&sip[4 * i], AF_INET, IPPROTO_UDP, "sip",
 				  SIP_PORT, ports[i], i, sip_exp_policy,
 				  SIP_EXPECT_MAX,
